modules/core/src/main/java/io/fluo/recipes/map/CollisionFreeMap.java

```diff
@@ -447,20 +456,37 @@ public Options(String mapId, String combinerType, String updateObserverType, Str
       this.updateObserverType = updateObserverType;
       this.keyType = keyType;
       this.valueType = valueType;
-      this.updateType = updateType;
     }
 
-    public <K, V, U> Options(String mapId, Class<? extends Combiner<K, V, U>> combiner,
-        Class<K> keyType, Class<V> valueType, Class<U> updateType, int buckets) {
-      this(mapId, combiner.getName(), keyType.getName(), valueType.getName(), updateType.getName(),
-          buckets);
+    /**
+     * Sets a limit on the amount of serialized updates to read into memory. Additional memory will
+     * be used to actually deserialize and process the updates. This limit does not account for
+     * object overhead in java, which can be significant.
+     */
+    public Options setBufferSize(long bufferSize) {
```

Comment:
From the code, it looks like `bufferSize` is in bytes and not the number of updates.  Can you make this more clear?